src/System.Threading/ref/System.Threading.cs

```diff
@@ -324,6 +330,24 @@ public partial class SynchronizationLockException : System.Exception
         public SynchronizationLockException() { }
         public SynchronizationLockException(string message) { }
         public SynchronizationLockException(string message, System.Exception innerException) { }
+        protected SynchronizationLockException(System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context) { }
+    }
+    public sealed partial class ThreadAbortException : System.SystemException
```

Comment:
>  It looks like there are dependencies on ThreadAbortException in CoreCLR native and managed code

We are not going to have Thread.Abort. The handling of ThreadAbortException in CoreCLR is  unrechable code - we should not need to worry about it.